paasta_tools/automatic_rollbacks.py

```diff
@@ -22,30 +23,98 @@ def get_slack_blocks_for_initial_deployment(message):
                 "text": message,
             },
         },
+        {"type": "divider"},
+        {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f"Status: {status}\nLast action: {last_action}",
+            },
+        },
         {
             "type": "actions",
-            "block_id": "rollback_block1",
-            "elements": [
+            "block_id": "deployment_actions",
+            "elements": get_button_elements(["rollback", "forward"], active_button=active_button),
+        },
+    ]
+    return blocks
+
+
+def get_button_elements(buttons, active_button=None):
+    elements = []
+    for button in buttons:
+        if button == "rollback" and active_button == "rollback":
+            elements.append(
+                {
+                    "type": "button",
+                    "text": {
+                        "type": "plain_text",
+                        "text": "Rolling Back :zombocom: (Not Implemented)",
+                        "emoji": True,
+                    },
+                    "value": "rollback",
+                },
+            )
+        elif button == "rollback" and active_button != "rollback":
```

Comment:
Multiple-page-long if-elif blocks make me nervous. I'd prefer this to use dicts:

```
buttonContentsActive = {
    "rollback": {
        "type": "button",
        "text": {
            "type": "plain_text",
            "text": "Rolling Back :zombocom: (Not Implemented)",
            "emoji": True,
        },
        "value": "rollback",
    },
    "forward":  {
        "type": "button",
        "text": {
            "type": "plain_text",
            "text": "Rolling Forward :zombocom: (Not Implemented)",
            "emoji": True,
        },
        "value": "forward",
    },
}
buttonContentsInactive = {
    "rollback": {
        "type": "button",
        "text": {
            "type": "plain_text",
            "text": "Roll Back :arrow_backward: (Not Implemented)",
            "emoji": True,
        },
        "value": "rollback",
        "confirm": get_confirmation_object("rollback"),
    },
    "forward": {
        "type": "button",
        "text": {
            "type": "plain_text",
            "text": "Continue Forward :arrow_forward: (Not Implemented)",
            "emoji": True,
        },
        "value": "continue",
        "value": "forward",
        "confirm": get_confirmation_object("forward"),
    },
}

if button == active_button:
    elements.append(buttonContentsActive[button])
else:
    elements.append(buttonContentsInactive[button])
```

I think it would be a lot harder to forget a case this way.

And yeah maybe DRY this a bit.

Reply:
Got it. I've un-nested the if blocks by using a dict and dried out the button creation (it does look better)